test/src/org/apache/jmeter/protocol/http/sampler/TestJmeterVariableSameUser.java

```diff
@@ -20,79 +20,76 @@
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
-import java.net.MalformedURLException;
-import java.net.URL;
-
 import org.apache.jmeter.protocol.http.control.CacheManager;
 import org.apache.jmeter.protocol.http.control.CookieManager;
 import org.apache.jmeter.protocol.http.control.gui.HttpTestSampleGui;
 import org.apache.jmeter.threads.JMeterContext;
 import org.apache.jmeter.threads.JMeterContextService;
-import org.apache.jmeter.threads.JMeterThread;
 import org.apache.jmeter.threads.JMeterVariables;
 import org.junit.Before;
 import org.junit.Test;
 
-public class TestHTTPHC4Impl {
-    public static String url="https://www.google.com";
+public class TestJmeterVariableSameUser {
     private JMeterContext jmctx;
     private JMeterVariables jmvars;
-    
     @Before
     public void setUp() {
-        jmctx = JMeterContextService.getContext();
-        jmctx.setVariables(new JMeterVariables());
-        jmvars = jmctx.getVariables();
+        jmctx = JMeterContextService.getContext();     
+        jmvars = new JMeterVariables();
     }
     @Test
-    public void testCookieManagerForDifferentUserOnSameIternation() throws MalformedURLException {
+    public void testCookieManagerForDifferentUserOnSameIternation() {
+        jmvars.putObject("__jmv_SAME_USER", true);
+        jmctx.setVariables(jmvars);        
         HTTPSamplerBase sampler = (HTTPSamplerBase) new HttpTestSampleGui().createTestElement();
-        jmvars.putObject(JMeterThread.IS_SAME_USER, false);
         CookieManager cookieManager = new CookieManager();
         cookieManager.setControlledByThread(true);
         sampler.setCookieManager(cookieManager);
-        HTTPHC4Impl hc = new HTTPHC4Impl(sampler);
-        hc.sample(new URL(url), "GET", true, 0);
+        sampler.setThreadContext(jmctx);
+        boolean res=(boolean) cookieManager.getThreadContext().getVariables().getObject("__jmv_SAME_USER");
         assertTrue("When test different user on the different iternation, the cookie should be cleared",
```

Comment:
Test is not correct, you just test variable is set and not that cookie is cleared